article/app/services/dotcomponents/ArticlePicker.scala

```diff
@@ -122,16 +122,45 @@ object ArticlePicker {
     isSupported
   }
 
-  def getTier(page: PageWithStoryPackage)(implicit request: RequestHeader): RenderType = {
+  def dcrShouldRender(request: RequestHeader): Boolean = {
+    // dcrShouldRender provides an override to let us force rendering by DCR even
+    // when an article is not supportted
+    val forceDCR = request.forceDCR
+    // our whitelist is only one article at the moment
+    val isInWhitelist = request.path == "/info/2019/dec/08/migrating-to-react"
+
+    forceDCR || isInWhitelist
+  }
 
-    val whitelistFeatures = featureWhitelist(page, request)
-    val isEnabled = conf.switches.Switches.DotcomRendering.isSwitchedOn
-    val isCommercialBetaUser = ActiveExperiments.isParticipating(DotcomRenderingAdvertisements)
+  def dcrShouldNotRender(request: RequestHeader): Boolean = {
+    val forceDCROff = request.forceDCROff
+    val dcrEnabled = conf.switches.Switches.DotcomRendering.isSwitchedOn
+
+    forceDCROff || !dcrEnabled
+  }
 
-    val tier = if ((dcrCouldRender(page, request) && isEnabled && isCommercialBetaUser && !request.forceDCROff) || request.forceDCR) RemoteRender else LocalRenderArticle
+  def getTier(page: PageWithStoryPackage)(implicit request: RequestHeader): RenderType = {
+    val whitelistFeatures = featureWhitelist(page, request)
+    val userIsInCohort = ActiveExperiments.isParticipating(DotcomRenderingAdvertisements)
+
+    // Decide if we should render this request with the DCR platform or not
+    // RemoteRender means DCR
+    val tier = if (dcrShouldNotRender(request)) {
+      // DCR is not turned on or dcr=false was passed
```

Comment:
Personally I wouldn't add these comments at this level. I think the method names would ideally be descriptive enough to give a high level understanding (which I think they are here), and if the reader wants to know more they can dive in to the method implementation.